### ISSUE TITLE
Issues with connecting using credentials: Access denied for user 

### ISSUE THREAD

**Issue by bonanza123** [mikilian/arm32v7-mariadb-alpine#1]
First of all: thanks for your images!
I hope you could help me with the following issue:

Here my `docker-compose.yml` which I am booting up with `docker-compose up -d`.

```
version: '2.1'
services:
    mysql-test:
      image: michaelkilian/arm32v7-mariadb-alpine:10.5.8
      stop_grace_period: 45s
      environment:
        - TZ=Europe/Berlin
        - MYSQL_ROOT_PASSWORD=rootpw
        - MYSQL_DATABASE=testdb
        - MYSQL_USER=mysqluser
        - MYSQL_PASSWORD=userpw
        - MYSQL_INITDB_SKIP_TZINFO=1
      restart: always
      ports:
        - 3306:3306
      networks:
        test-network:
          aliases:
            - mysql

networks:
  test-network:
    driver: bridge
    driver_opts:
      com.docker.network.bridge.name: br-test
    enable_ipv6: true
    ipam:
      driver: default
      config:
        - subnet: 172.123.1.0/24
        - subnet: fd4d:6169:6c63:6f77::/64
 ```

Then I am trying to connect using
```
docker-compose exec mysql-test mysql -umysqluser -puserpw testdb
```

which gives me:
`ERROR 1045 (28000): Access denied for user 'mysqluser'@'localhost' (using password: YES)`

Here an except from the server log:

> # docker-compose up  
> Creating network "mailcowdockerized_test-network" with driver "bridge"
> Creating mailcowdockerized_mysql-test_1 ... done
> Attaching to mailcowdockerized_mysql-test_1
> mysql-test_1  | [i] mysqld not found, creating....
> mysql-test_1  | [i] MySQL data directory not found, creating initial DBs
> mysql-test_1  | [i] Creating database: testdb
> mysql-test_1  | [i] with character set: 'utf8' and collation: 'utf8_general_ci'
> mysql-test_1  | [i] Creating user: mysqluser with password userpw
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] /usr/bin/mysqld (mysqld 10.4.15-MariaDB) starting as process 71 ...
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Using Linux native AIO
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Uses event mutexes
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Number of pools: 1
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Using generic crc32 instructions
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Completed initialization of buffer pool
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Creating shared tablespace for temporary tables
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: 10.4.15 started; log sequence number 60967; transaction id 21
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] Plugin 'FEEDBACK' is disabled.
> mysql-test_1  | 2020-12-24 14:40:10 0 [Note] InnoDB: Buffer pool(s) load completed at 201224 14:40:10
> mysql-test_1  | 2020-12-24 14:40:10 6 [Warning] 'user' entry '@03504deb188a' ignored in --skip-name-resolve mode.
> mysql-test_1  | 2020-12-24 14:40:10 6 [Warning] 'proxies_priv' entry '@% root@03504deb188a' ignored in --skip-name-resolve mode.
> mysql-test_1  | 2020-12-24 14:40:10 6 [Warning] 'user' entry '@03504deb188a' ignored in --skip-name-resolve mode.
> mysql-test_1  | 2020-12-24 14:40:10 6 [Warning] 'proxies_priv' entry '@% root@03504deb188a' ignored in --skip-name-resolve mode.
> mysql-test_1  | /scripts/run.sh: ignoring or entrypoint initdb empty /docker-entrypoint-initdb.d/*
> mysql-test_1  | 
> mysql-test_1  | 
> mysql-test_1  | MySQL init process done. Ready for start up.
> mysql-test_1  |
> mysql-test_1  | exec /usr/bin/mysqld --user=mysql --console --skip-name-resolve --skip-networking=0
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] /usr/bin/mysqld (mysqld 10.4.15-MariaDB) starting as process 1 ...
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Using Linux native AIO
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Uses event mutexes
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Number of pools: 1
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Using generic crc32 instructions
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Completed initialization of buffer pool
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Creating shared tablespace for temporary tables
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: 10.4.15 started; log sequence number 60976; transaction id 21
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] Plugin 'FEEDBACK' is disabled.
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] InnoDB: Buffer pool(s) load completed at 201224 14:40:13
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] Server socket created on IP: '::'.
> mysql-test_1  | 2020-12-24 14:40:13 0 [Warning] 'user' entry '@03504deb188a' ignored in --skip-name-resolve mode.
> mysql-test_1  | 2020-12-24 14:40:13 0 [Warning] 'proxies_priv' entry '@% root@03504deb188a' ignored in --skip-name-resolve mode.
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] Reading of all Master_info entries succeeded
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] Added new Master_info '' to hash table
> mysql-test_1  | 2020-12-24 14:40:13 0 [Note] /usr/bin/mysqld: ready for connections.
> mysql-test_1  | Version: '10.4.15-MariaDB'  socket: '/run/mysqld/mysqld.sock'  port: 3306  MariaDB Server
> ^[[A^[[Amysql-test_1  | 2020-12-24 14:46:49 8 [Warning] Access denied for user 'mysqluser'@'localhost' (using password: YES)

**Comment by bonanza123** [comment]
I gues there is a problem with granting the priveleges to the user. 

I tried the following:

> # docker-compose exec mysql-test sh
> / # mysql -uroot -prootpw
> Welcome to the MariaDB monitor.  Commands end with ; or \g.
> Your MariaDB connection id is 10
> Server version: 10.4.15-MariaDB MariaDB Server
> 
> Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.
> 
> Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
> 
> MariaDB [(none)]> SELECT user FROM mysql.user;
> +-------------+
> | User        |
> +-------------+
> | mysqluser   |
> | root        |
> |             |
> |             |
> | mariadb.sys |
> | mysql       |
> | root        |
> +-------------+
> 7 rows in set (0.004 sec)
> 
> MariaDB [(none)]> SHOW GRANTS FOR 'mysqluser'@'localhost';
> ERROR 1141 (42000): There is no such grant defined for user 'mysqluser' on host 'localhost'
> MariaDB [(none)]> SHOW GRANTS FOR 'root'@'localhost';
> +----------------------------------------------------------------------------------------------------------------------------------------+
> | Grants for root@localhost                                                                                                              |
> +----------------------------------------------------------------------------------------------------------------------------------------+
> | GRANT ALL PRIVILEGES ON *.* TO `root`@`localhost` IDENTIFIED BY PASSWORD '*79D0CF9A6A052105DA1E1181406C34FC87AAC89D' WITH GRANT OPTION |
> | GRANT PROXY ON ''@'%' TO 'root'@'localhost' WITH GRANT OPTION                                                                          |
> +----------------------------------------------------------------------------------------------------------------------------------------+
> 2 rows in set (0.001 sec)

**Comment by mikilian** [comment]
I was able to reproduce and fix this issue (f7f2cd5).

```bash
$ docker-compose exec test mysql -utest -ptestpw testdb
Welcome to the MariaDB monitor.  Commands end with ; or \g.                                                          
Your MariaDB connection id is 8
Server version: 10.4.15-MariaDB MariaDB Server

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.                                                 

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.                                       

MariaDB [testdb]> SHOW DATABASES;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| testdb             |
+--------------------+
2 rows in set (0.003 sec)

MariaDB [testdb]> CREATE TABLE test (blob_col BLOB, INDEX(blob_col(10)));
Query OK, 0 rows affected (0.022 sec)

MariaDB [testdb]> SELECT * from test;
Empty set (0.005 sec)

MariaDB [testdb]> SHOW TABLES;
+------------------+
| Tables_in_testdb |
+------------------+
| test             |
+------------------+
1 row in set (0.002 sec)

MariaDB [testdb]>

```

Unfortunately, I noticed that alpine's version 3.12 supports MariaDB 10.4.15 at most. Since there are currently problems with our internet provider, I can't reach the edge repository and thus can't offer a current build of 10.5. As soon as I can get this back up, I'll push it to the docker hub after the fact.

Tag 10.5.8 has thus been removed from the git and docker hub, 10.4.15 was rebuilt and set to latest.

Merry Christmas - even if late - and have a nice evening.
